robot_smach_states/src/robot_smach_states/startup.py

```diff
@@ -239,3 +239,59 @@ def execute(self, userdata=None):
 
         rospy.loginfo("Timed out with door still closed")
         return "closed"
+
+
+class WaitForLocalPlanner(smach.State):
+    """
+    Wait till a valid footprint message from the local planner is received. A footprint is valid if it contains at least
+    3 points. If no valid message is received within the timeout, "timeout" is returned.
+    """
+    def __init__(self, robot, timeout):
+        """
+        Constructor
+        :param robot: robot object
+        :type robot: robot
+        :param timeout: timeout
+        :type timeout: (float, int)
+        """
+        smach.State.__init__(self, outcomes=["ready", "timeout"])
+        self._robot = robot
+        self._timeout = timeout
+
+        self._local_planner_ready = Event()
```

Comment:
In my previous review I left a commented that didn't come through: I think you can get rid of the local planner ready and footprint sub member attributes. Concerning the subscriber: you also have to unregister upon fail, so why do it in the callback as well?

The event is a bit more challenging but could also work:
```python
def msg_cb(self, ready_event, msg):
    try:
        if len(msg.polygon.points) >= 3:
            rospy.loginfo("Valid footprint received")
            ready_event.set()
    except Exception as e:
        pass

def execute(self, userdate=None):
    rospy.loginfo("Waiting for local planner footprint")
    footprint_topic = "/{}/local_planner/local_costmap/robot_footprint/footprint_stamped".\
        format(self._robot.robot_name)

    ready_event = Event()
    footprint_sub = rospy.Subscriber(footprint_topic, PolygonStamped, functools.partial(self.msg_cb, ready_event))

    ready_before_timeout = self._local_planner_ready.wait(self._timeout)

    rospy.loginfo("Unregistering footprint subscriber")
    footprint_sub.unregister()

    if ready_before_timeout:
        rospy.loginfo("Local Planner is ready")
        return "ready"

    rospy.loginfo("Local Planner not ready during timeout")
    return "timeout"
```
Check out, e.g., https://answers.ros.org/question/39774/subscriber-callback-function-with-multiple-parameters/

Finally, I don't see why the try-except block is necessary